### DISCUSSION TITLE
Comparison with xournalpp

### DISCUSSION THREAD

**Discussion by lamyergeier** [flxzt/rnote#416] [Q&A] [answered]
Could you please tell me pros and cons as compared with [xournalpp/xournalpp](https://github.com/xournalpp/xournalpp). Why should we use this software over xournalpp?

**Comment by benjamin051000** [comment]
I've been using Xournal++ for a little while, and Rnote has only recently caught my eye.

**Reasons to use Xournal++:**
- Xournal++ has been around for much longer than Rnote, and in that regard tends to be more stable.
- Xournal++ has a pdf annotation mode which is sorely needed in Rnote (see #153 )

**Reasons to use Rnote:**
- Much prettier user interface than Xournal++ (X++ feels dated)
- Rnote seems to have better pen support (e.g., pen alternate buttons), or in my experience they can be trickier to set up in Xournal++

Overall I'd say try both and see what you like.

**Comment by mizhozan** [comment, marked as answer]
To me it was amazing, how a rust based solution could cover almost much xournalpp can do and even better in many aspect. Rnote is sleep and even run smoother. Xournalpp, has been working on its next generation xml file format to include media in the file, but rnote has that feature just out of the box at early releases. I still have both at hands and try to support both, but I have been gravitated more and more toward Rnote in recent weeks.

Plus, rnote supports infinite canvas and better stylus support on top of better UI.

This being said, xournalpp has been around for longer time and it's a bit more mature. Also, at least currently, it has better PDF annotation support.

**Comment by bittner** [comment]
I figured that more people like me are looking for a note taking app that works like a tool that can add visual annotations on somewhat "any" document (which usually boils down to PDF, because modern word processors have annotating built in). That's where the run on Xournal++ comes from, I believe.

Xournal++ handles PDF as an external document (at least for now), also based on the fact that a PDF is meant to be the document's background image, which makes it a bit cumbersome to work with (PDF editing features are not "first-class citizens") and causes issues when moving documents in the file system.

I'm a bit worried that Rnote will fall into the same trap. For now, the pleasant editing experience makes the difference, but in the long run a "[don't make me think](https://archive.org/details/dontmakemethinkr0000krug/mode/2up)" :tm:  experience for opening, manipulating, saving files might be critical for user adoption. (See https://github.com/xournalpp/xournalpp/issues/6381 for my related proposal on the Xournal++ issue tracker.)

**Comment by SimonBauer-git** [comment]
rnote is just more modern looking, and the ui is less glitchy. also thank you guys for infinite paper, individual pages just dont fit my Workflow anymore.
